dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task221007AddVariantIntoPrimaryKey.java

```diff
@@ -26,31 +27,30 @@ public boolean forceRun() {
     }
 
     public String getPostgresScript() {
-        return getScript("contentlet_version_info_pkey");
+        return getScript();
     }
 
 
     public String getMSSQLScript(){
+        return getScript();
+    }
+
+    public String getScript() {
+        final StringBuffer buffer = new StringBuffer();
 
-        try {
-            final ArrayList arrayList = new DotConnect()
-                    .setSQL("SELECT * FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS T "
-                            + "WHERE table_name = 'contentlet_version_info' "
-                            + "AND constraint_type = 'PRIMARY KEY'")
-                    .loadResults();
-
-            final String constraintName = (((Map) arrayList.get(0))).get("constraint_name").toString();
-            return getScript(constraintName);
-        } catch (DotDataException e) {
-            throw new DotRuntimeException(e);
+        final Optional<String> contentletVersionInfoOptional = DotDatabaseMetaData.getPrimaryKeyName(
```

Comment:
Optional for the win!!
This can also be simplified as:
```
DotDatabaseMetaData.getPrimaryKeyName("contentlet_version_info")
  ifPresent(version ->  buffer.append(String.format("ALTER TABLE contentlet_version_info DROP CONSTRAINT %s;", version)));
```

Reply:
+1

Reply:
but ifPresent return void and I need the value later